File: backend-java/src/test/java/com/financepulse/engine/testsupport/FakeTokenService.java
package com.financepulse.engine.testsupport;

import com.financepulse.engine.application.ports.TokenService;
import java.util.Optional;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class FakeTokenService implements TokenService {

    private static final Pattern TOKEN_PATTERN = Pattern.compile("^token-for-(.+)$");

    @Override
    public String issue(String userId) {
        return "token-for-" + userId;
    }

    @Override
    public Optional<String> verify(String token) {
        Matcher matcher = TOKEN_PATTERN.matcher(token);
        return matcher.matches() ? Optional.of(matcher.group(1)) : Optional.empty();
    }
}

File: backend/src/test-support/FakeTokenService.ts
import type { TokenPayload, TokenService } from "../application/ports/TokenService";

export class FakeTokenService implements TokenService {
  issue(userId: string): string {
    return `token-for-${userId}`;
  }

  verify(token: string): TokenPayload | null {
    const match = /^token-for-(.+)$/.exec(token);
    return match ? { userId: match[1] } : null;
  }
}
